lib/SharedBuffer.h

```diff
@@ -203,6 +203,13 @@ class SharedBuffer {
         writeIdx_ = 0;
     }
 
+    bool pop(std::string& target) {
+        if (data_ == nullptr) return false;
```

Comment:
The returned value is always ignored. I think we should throw an exception to disable `moveDataIntoString` on an invalid `Message` object, i.e. built from a `MessageBuilder` whose content was set by `setAllocatedContent` .

Reply:
Good point. Let me add a flag in `SharedBuffer` to denote if it's a read-only one.

Reply:
Good point.

Reply:
Done